paddle/fluid/operators/math/detail/lstm_cpu_kernel.h

```diff
@@ -227,31 +229,33 @@ void avx_lstm_backward_one_sequence(Op op, LstmMetaValue<T> value,
   __m256 r_checkFGrad;
   __m256 r_checkOGrad;
 
-  __m256 *value_in = (__m256 *)value.gate_value;
-  __m256 *value_ig = (__m256 *)(value.gate_value + frame_size);
-  __m256 *value_fg = (__m256 *)(value.gate_value + frame_size * 2);
-  __m256 *value_og = (__m256 *)(value.gate_value + frame_size * 3);
-  __m256 *grad_in = (__m256 *)grad.gate_grad;
-  __m256 *grad_ig = (__m256 *)(grad.gate_grad + frame_size);
-  __m256 *grad_fg = (__m256 *)(grad.gate_grad + frame_size * 2);
-  __m256 *grad_og = (__m256 *)(grad.gate_grad + frame_size * 3);
+  __m256 *value_in = reinterpret_cast<__m256>(value.gate_value);
+  __m256 *value_ig = reinterpret_cast<__m256>(value.gate_value + frame_size);
+  __m256 *value_fg =
+      reinterpret_cast<__m256>(value.gate_value + frame_size * 2);
+  __m256 *value_og =
+      reinterpret_cast<__m256>(value.gate_value + frame_size * 3);
+  __m256 *grad_in = reinterpret_cast<__m256> grad.gate_grad;
```

Comment:
I think you missed the paranthesis here. (probably should be`reinterpret_cast<__m256>(grad.gate_grad)`);

Reply:
Done. Please review again.